src/goto-programs/simplify_function_pointers.cpp

```diff
@@ -0,0 +1,49 @@
+//
+// Created by hannes on 22/10/2019.
+//
+
+#include "simplify_function_pointers.h"
+#include <goto_model.h>
+#include <util/fresh_symbol.h>
+#include <util/simplify_expr.h>
+
+void simplify_function_pointers(goto_modelt &goto_model)
+{
+  for(auto &goto_function : goto_model.goto_functions.function_map)
+  {
+    for_each_goto_location(
+      goto_function.second, [&](goto_programt::targett it) {
+        if(it->is_function_call())
```

Comment:
Not sure it would help much, but you could use the `for_each_goto_location_if` and split the `is_function_call /\ is_dereference` as `Predicate`.